### PR TITLE
[csharp] Introducing the `enumPropertyNaming` option to the C# Generator

### PR DESCRIPTION
**Pull request by fujieda**
This pull request introduces the `enumPropertyNaming` option to the C# generator. This option, which is currently supported by the Kotlin and TypeScript generators, allows you to control the naming style of enum names.

In our projects, we consistently use `UPPER_SNAKE_CASE` for constants in our OpenAPI definitions, and we prefer to maintain this naming style in our C# code. However, the current C# generator changes the enum names by removing the underscores, resulting in enum names like `UPPERSNAKECASE`.

While searching for a solution to this problem, I came across the `enumPropertyNaming` option available in the Kotlin and TypeScript generators. To resolve this discrepancy and ensure that our enum naming style remains consistent with `UPPER_SNAKE_CASE`, I ported this option to the C# generator.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.1.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

@mandrean @shibayan @Blackclaws @lucamazzanti @iBicha

### CODE REVIEW

**Comment by wing328** [comment]
@fujieda thanks for the PR.

What about adding an option similar to `modelNameMapping`, `parameterNameMapping` such as `enumNameMapping` to have complete control of the enum naming in the C# generator?

ref: https://github.com/OpenAPITools/openapi-generator/blob/master/docs/customization.md#name-mapping

**Comment by fujieda** [comment]
@wing328

I couldn't find an option to define enum name mappings similar to `modelNameMapping` and `parameterNameMapping` in the C# generator. Upon reviewing the `toEnumVarName` method in `AbstractCsharpCodegen`, it appears that there's no code implementation for applying name mappings, unlike `toVarName`, `toParamName`, and `toModelName`, which support such mappings.

However, in our projects, our primary goal is not to establish specific name mappings, but rather to exert control over the naming styles of enum names. We have a significant number of enum names within our OpenAPI definitions, over fifty in total. Given this scale, defining individual name mappings for each enum name isn't a practical or feasible approach for us.

**Comment by wing328** [comment]
> I couldn't find an option to define enum name mappings similar to modelNameMapping and parameterNameMapping in the C# generator.

Sorry I wasn't clear. I mean "adding an option" to your requirement (which means the option doesn't exist yet)

Thanks for sharing the use cases. I'll review and get back to you.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java [hunk 1]
```diff
@@ -1644,6 +1666,27 @@ public String toEnumVarName(String name, String datatype) {
         }
     }
 
+    private String adjustNamingStyle(String name)
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:1669]: minor suggestion: add a docstring explaining what this function does.

Comment (fujieda) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractCSharpCodegen.java:1675]: I have added a doc comment.

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/csharpnetcore/CSharpModelEnumTest.java [hunk 2]
```diff
@@ -148,4 +152,54 @@ public void useEmptyEnumSuffixes() {
...
+    @DataProvider(name = "enumVarName")
+    public Object[][] provideTestData()
+    {
+        return new Object[][] {
+            { "FooBar", "fooBar", camelCase },
+            { "fooBar", "fooBar", camelCase },
+            { "foo-bar", "fooBar", camelCase },
+            { "foo_bar", "fooBar", camelCase },
+            { "foo bar", "fooBar", camelCase },
+            { "FOO-BAR", "fOOBAR", camelCase }, // camelize doesn't support uppercase
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/csharpnetcore/CSharpModelEnumTest.java:173]: we will try to fix that in another PR.

**Comment by wing328** [comment]
LGTM. Let's give it a try.

Thanks again for the PR.

**Pull request merged**